core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -37,8 +37,30 @@
 abstract class BaseMetadataTable implements Table, Serializable {
   private final PartitionSpec spec = PartitionSpec.unpartitioned();
   private final SortOrder sortOrder = SortOrder.unsorted();
+  private final TableOperations ops;
+  private final Table table;
+  private final String name;
 
-  abstract Table table();
+  protected BaseMetadataTable(TableOperations ops, Table table, String name) {
```

Comment:
should this also implement "hasTableOperations"  then?

Reply:
I can also understand if we don't ops protected and not public

Reply:
Yeah, I actually implemented `HasTableOperations` in both metadata and txn tables. I think that's a better alternative.